### PR TITLE
Fix MessageBufferU#copyTo called by sliced buffer

### PR DESCRIPTION
**Pull request by komamitsu**
Fix https://github.com/msgpack/msgpack-java/issues/282

### CODE REVIEW

**Comment by komamitsu** [comment]
@xerial Can you take a look at this pull request?

**Comment by komamitsu** [comment]
https://travis-ci.org/msgpack/msgpack-java/jobs/87210273 MessageBufferBE can't be used in Travis CI...?

**Comment by xerial** [comment]
right. MessageBufferBE only works in big endian machine. And initializing MessageBufferU and MessageBufferU in the code slows down the performance of MessageBuffer performance test. Use reflection to instantiate MessageBufferU objects.

File: msgpack-core/src/test/scala/org/msgpack/core/buffer/MessageBufferTest.scala [hunk 1]
```diff
@@ -184,6 +184,47 @@ class MessageBufferTest
...
+        dstBuffer.getByte(1) shouldBe 0x05 // copied by sliced.getByte(3)
+        dstBuffer.getByte(2) shouldBe 0x06 // copied by sliced.getByte(4)
+        dstBuffer.getByte(3) shouldBe 0x03
+        dstBuffer.getByte(4) shouldBe 0x04
+        dstBuffer.getByte(5) shouldBe 0x05
+        dstBuffer.getByte(6) shouldBe 0x06
+        dstBuffer.getByte(7) shouldBe 0x07
+      }
+
+      checkSliceAndCopyTo(new MessageBufferU(prepareBytes), new MessageBufferU(prepareBytes))
```
Comment (xerial) [inline msgpack-core/src/test/scala/org/msgpack/core/buffer/MessageBufferTest.scala]: MessageBufferU should not be called directly. This changes every MessageBuffer method call to invokeinterface, and deteriorates the performance. TravisCI uses MessageBufferU for jvm6, I think.

Comment (xerial) [inline msgpack-core/src/test/scala/org/msgpack/core/buffer/MessageBufferTest.scala]: Sorry. For some reason, jvm6 test was turned off. Run test with -Dmsgpack.universal-buffer=true

Comment (xerial) [inline msgpack-core/src/test/scala/org/msgpack/core/buffer/MessageBufferTest.scala]: https://github.com/msgpack/msgpack-java/blob/v07-develop/.travis.yml#L14
I found we are tresting MessageBufferU here, so having a test code that copies data between buffers is sufficient.

Comment (komamitsu) [inline msgpack-core/src/test/scala/org/msgpack/core/buffer/MessageBufferTest.scala]: Good to hear. I'll change the test code later. Thanks.

Comment (komamitsu) [inline msgpack-core/src/test/scala/org/msgpack/core/buffer/MessageBufferTest.scala]: @xerial Fixed the test code. Can you review it again?

**Comment by xerial** [comment]
Merged. Thanks!

**Pull request merged**